src/garbage_collector.rs

```diff
@@ -219,40 +249,128 @@ impl GarbageCollectorOrchestrator {
     }
 
     /// Run the garbage collector
-    pub async fn run(&self) {
-        let manifest_ticker = Self::options_to_ticker(self.options.manifest_options.as_ref());
-        let wal_ticker = Self::options_to_ticker(self.options.wal_options.as_ref());
-        let compacted_ticker = Self::options_to_ticker(self.options.compacted_options.as_ref());
+    pub async fn run(&self, shutdown_tx: watch::Sender<bool>) {
+        let log_ticker = crossbeam_channel::tick(Duration::from_secs(60));
+        let (manifest_ticker, mut manifest_status) =
+            Self::options_to_ticker(self.options.manifest_options.as_ref());
+        let (wal_ticker, mut wal_status) =
+            Self::options_to_ticker(self.options.wal_options.as_ref());
+        let (compacted_ticker, mut compacted_status) =
+            Self::options_to_ticker(self.options.compacted_options.as_ref());
+
+        info!(
+            "Starting Garbage Collector with [manifest: {}], [wal: {}], [compacted: {}]",
+            manifest_status, wal_status, compacted_status
+        );
 
         loop {
             crossbeam_channel::select! {
+                recv(log_ticker) -> _ => {
+                   info!("GC has collected {} Manifests, {} WAL SSTs and {} Compacted SSTs.",
+                        self.db_stats.gc_manifest_count.value.load(Ordering::SeqCst),
+                        self.db_stats.gc_wal_count.value.load(Ordering::SeqCst),
+                        self.db_stats.gc_compacted_count.value.load(Ordering::SeqCst)
+                    );
+                },
                 recv(manifest_ticker) -> _ => {
+                    debug!("Scheduled garbage collection attempt for Manifests.");
                     if let Err(e) = self.collect_garbage_manifests().await {
                         error!("Error collecting manifest garbage: {}", e);
                     }
-                }
+                    if manifest_status == DirGcStatus::OneMore { manifest_status = DirGcStatus::Done }
```

Comment:
you could pull this into a fn in `DirGcStatus` like:
```
impl DirGcStatus {
    fn advance(&mut self) {
        let next = match self {
            DirGcStatus::Indefinite(d) => Indefinite(*d),
            DirGcStatus::OneMore => Done,
            DirGcStatus::Done => Done
        };
        *self = next;
    }
}
```

and then just call `manifest_status.advance()`